### PR TITLE
Add RealtimeRequest marker interface to group realtime operations together

### PR DESCRIPTION
**Pull request by martijnvg**
The following classes will implement it: GetRequest, MultiGetRequest, TermVectorsRequest and MultiTermVectorsRequest..

### CODE REVIEW

**Comment by jaymode** [comment]
LGTM

**Comment by nik9000** [comment]
Maybe add the ```@Override``` annotation on the methods that implement that interface?

File: core/src/main/java/org/elasticsearch/action/RealtimeRequest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Indicates that a request can execute in realtime (reads from the translog).
+ * All {@link ActionRequest} that are realtime should implement this interface.
+ */
+public interface RealtimeRequest {
+
+    /**
+     * @param realtime Controls whether this request should be realtime by reading from the translog.
+     */
+    <R extends RealtimeRequest> R realtime(Boolean realtime);
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/RealtimeRequest.java]: Can/should this have a consistent default?

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/RealtimeRequest.java]: Also an explanation for what a null parameter means in this context would be cool.

**Comment by martijnvg** [comment]
@nik9000 I'll add the `@Override` annotation.

File: core/src/main/java/org/elasticsearch/action/RealtimeRequest.java (hunk 1 above)
Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/action/RealtimeRequest.java]: hmmm... in all cases where null is being normalised to true. So instead documenting what `null` mean we should simplify and change `realtime` field in all places to be a primitive boolean and default to `true`.

Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/action/RealtimeRequest.java]: I forgot that the TransportGetAction and TransportShardMultiGetAction  actually do check if `realtime` is set to `null` and then use a default (based on `action.get.realtime` setting). But this isn't the case for the term vector and multi term vector apis... I wonder if this default mechanism is actually being used. I think we should remove it, it cleans up the code and simplifies.

**Comment by martijnvg** [comment]
@nik9000 I updated the PR. About the realtime default, I think the `realtime` field should become a primitive boolean, it should just default to `true` everywhere (it is doing this already `action.get.realtime` defaults to `true`) and we should `remove` the `action.get.realtime` defaulting mechanism. But I think this should be done in a different change.

**Comment by nik9000** [comment]
Fair enough. LGTM.

**Pull request merged**